test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/ClientYamlTestSuite.java

```diff
@@ -139,39 +142,45 @@ public TeardownSection getTeardownSection() {
     }
 
     public void validate() {
-        validateExecutableSections(this, setupSection.getExecutableSections(), null, setupSection, null);
-        validateExecutableSections(this, teardownSection.getDoSections(), null, null, teardownSection);
+        Stream<String> errors = validateExecutableSections(setupSection.getExecutableSections(), null, setupSection, null);
+        errors = Stream.concat(errors, validateExecutableSections(teardownSection.getDoSections(), null, null, teardownSection));
         for (ClientYamlTestSection testSection : testSections) {
-            validateExecutableSections(this, testSection.getExecutableSections(), testSection, setupSection, teardownSection);
+            errors = Stream.concat(errors,
```

Comment:
This is perfectly fine as it is. Just for the fun of it, could be written as 
```
testSections.stream()
    .flatMap(testSection -> validateExecutableSections(testSection.getExecutableSections(), testSection, setupSection, teardownSection)) )
```

Reply:
good one, thanks :)